### ISSUE TITLE
[2.84] Syslog flooded with "IPv4: martian source" messages

### ISSUE THREAD

**Issue by iliv** [transmission/transmission#3341]
I run older transmission-daemon 2.84 (14307) and I'm getting a lot of these messages in syslog when Transmission is running, doesn't matter if it's transmission-daemon or transmission-gtk:

```
$ dmesg -w
[2346719.431459] IPv4: martian source 172.16.16.252 from 172.16.16.252, on dev ens1
[2346719.431478] ll header: 00000000: 00 0a cd 26 1c 49 60 a4 4c 26 f5 88 08 00        ...&.I`.L&....
[2346719.959125] IPv4: martian source 172.16.16.252 from 172.16.16.252, on dev ens1
[2346719.959143] ll header: 00000000: 00 0a cd 26 1c 49 60 a4 4c 26 f5 88 08 00        ...&.I`.L&....
[2346720.966976] IPv4: martian source 172.16.16.252 from 172.16.16.252, on dev ens1
[2346720.967025] ll header: 00000000: 00 0a cd 26 1c 49 60 a4 4c 26 f5 88 08 00        ...&.I`.L&....
[2346722.985525] IPv4: martian source 172.16.16.252 from 172.16.16.252, on dev ens1
[2346722.985573] ll header: 00000000: 00 0a cd 26 1c 49 60 a4 4c 26 f5 88 08 00        ...&.I`.L&....
[2346727.019624] IPv4: martian source 172.16.16.252 from 172.16.16.252, on dev ens1
[2346727.019671] ll header: 00000000: 00 0a cd 26 1c 49 60 a4 4c 26 f5 88 08 00        ...&.I`.L&....
[2346735.069873] IPv4: martian source 172.16.16.252 from 172.16.16.252, on dev ens1
[2346735.069925] ll header: 00000000: 00 0a cd 26 1c 49 60 a4 4c 26 f5 88 08 00        ...&.I`.L&....
[2346748.043008] IPv4: martian source 172.16.16.252 from 172.16.16.252, on dev ens1
[2346748.043027] ll header: 00000000: 00 0a cd 26 1c 49 60 a4 4c 26 f5 88 08 00        ...&.I`.L&....
[2346748.848177] IPv4: martian source 172.16.16.252 from 172.16.16.252, on dev ens1
[2346748.848194] ll header: 00000000: 00 0a cd 26 1c 49 60 a4 4c 26 f5 88 08 00        ...&.I`.L&....
[2346748.980057] IPv4: martian source 172.16.16.252 from 172.16.16.252, on dev ens1
[2346748.980095] ll header: 00000000: 00 0a cd 26 1c 49 60 a4 4c 26 f5 88 08 00        ...&.I`.L&....
[2346751.836610] IPv4: martian source 172.16.16.252 from 172.16.16.252, on dev ens1
[2346751.836629] ll header: 00000000: 00 0a cd 26 1c 49 60 a4 4c 26 f5 88 08 00        ...&.I`.L&....
[2346769.868627] IPv4: martian source 172.16.16.252 from 172.16.16.252, on dev ens1
[2346769.868648] ll header: 00000000: 00 0a cd 26 1c 49 60 a4 4c 26 f5 88 08 00        ...&.I`.L&....
^C
```

transmission-daemon is configured to listen for incoming connections on 172.16.16.252:50252.

```
$ sudo tcpdump -ni ens1 src 172.16.16.252 and dst 172.16.16.252   
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode    
listening on ens1, link-type EN10MB (Ethernet), capture size 262144 bytes     
10:29:29.110241 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 290 
10:29:29.645849 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 290 
10:29:30.721400 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 290 
10:29:32.770880 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 290 
10:29:36.786175 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 290 
10:29:44.791553 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 290 
10:29:57.692371 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 290 
10:29:58.624651 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 20  
10:29:58.759567 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 20  
10:30:01.681522 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 20  
10:30:19.712782 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 20  
12:29:29.881166 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 502 
12:29:30.408853 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 502 
12:29:31.416704 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 502 
12:29:33.435253 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 502 
12:29:37.469349 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 502 
12:29:45.519588 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 502 
12:29:58.492720 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 502 
12:29:59.297911 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 20  
12:29:59.429791 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 20  
12:30:02.286306 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 20  
12:30:20.318342 IP 172.16.16.252.50252 > 172.16.16.252.50252: UDP, length 20  
```

Why is Transmission talking to itself in such a peculiar fashion? What is it trying to do here?

**Comment by iliv** [comment]
Just to clarify, this traffic is clearly generated by Transmission and Transmission only. As soon as I stop Transmission the "martian source" messages go away and never are generated by anything else on the system.

**Comment by iliv** [comment]
One more thing.

Per RFC1812 section "5.3.7 Martian Address Filtering":

```
If a router discards a packet because of these rules, it SHOULD log at least the IP source address, 
the IP destination address, and, if the problem was with the source address, the physical interface
on which the packet was received  and the Link Layer address of the host or router from which the
packet was received.
```

The Link Layer address in the martian source messages matches MAC address of my ens1 interface:

```
$ ip l l ens1
2: ens1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP mode DEFAULT group default qlen 1000
    link/ether 00:0a:cd:26:1c:49 brd ff:ff:ff:ff:ff:ff
```

This echoes what tcpdump is showing – transmission appears to be sending traffic to itself from and to the same interface using exactly same ip:port as source and destination.

**Comment by ckerr** [comment]
2.84 came out in 2014 and we're working on getting 4.0.0-beta.1 out the door right now.

Could you test this out on a newer build and see if the problem persists?

**Comment by iliv** [comment]
Unfortunately, same problem in 3.00

**Comment by ckerr** [comment]
Again, I'd ask for testing with something from the 4 line :smile_cat:

**Comment by ckerr** [comment]
Looking at this issue again, I believe this is a duplicate of #295 and is fixed in `4.0.0-beta.1` and higher by #2756.
